### PR TITLE
UCP: add timeout config for query metrics from Prometheus

### PR DESCRIPTION
**Pull request by hsqlu**
Signed-off-by: Qiannan Lyu <lvqiannan@gmail.com>

UCP: #1753 

### What problem does this PR solve? 

fixes: #1753 

### What is changed and how does it work?

Added timeout context for http request and set the timeout time 5s. 

### Check List 

Tests 

Code changes

 - Has Go code change

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/autoscaler/autoscaler/calculate/calculate.go [hunk 1]
```diff
@@ -46,7 +48,10 @@ type SingleQuery struct {
 func queryMetricsFromPrometheus(tac *v1alpha1.TidbClusterAutoScaler, client promClient.Client, sq *SingleQuery, resp *Response) error {
 	query := sq.Quary
 	timestamp := sq.Timestamp
-	req, err := http.NewRequest("GET", fmt.Sprintf("%s%s", sq.Endpoint, queryPath), nil)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
```
Comment (DanielZhangQD) [inline pkg/autoscaler/autoscaler/calculate/calculate.go:52]: Suggesting defining a variable for `5` here.

Comment (hsqlu) [reply pkg/autoscaler/autoscaler/calculate/calculate.go:52]: How about define it as a constant out of this function?

Comment (DanielZhangQD) [reply pkg/autoscaler/autoscaler/calculate/calculate.go]: Great!

**Comment by hsqlu** [comment]
@Yisaer  PTAL. Thanks

**Pull request merged**